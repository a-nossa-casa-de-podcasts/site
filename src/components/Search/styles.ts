import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.blue};
    height: 100vh;
    overflow: hidden;
    padding: ${theme.spacings.small};
    padding-bottom: 0;
    display: grid;
    grid-template-rows: 60px auto minmax(10px, 60px);
    ${media.greaterThan('medium')`
      grid-template-rows: 100px auto minmax(10px, 60px);
      padding: ${theme.spacings.large};
    `}
  `}
`

export const SearchInputWrapper = styled.div`
  position: relative;
  width: 100%;
`

export const SearchInput = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    input {
      font-size: ${theme.font.sizes.xsmall};
      font-weight  ${theme.font.normal};
      font-family: ${theme.font.family};
      width: 100%;
      background: transparent;
      color: ${theme.colors.white};
      border: none;
      border-bottom: 2px solid white;
      padding: ${theme.spacings.xxsmall};
      padding-left: 0;
      outline: none;
      text-indent: 5rem;
      ${media.greaterThan('medium')`
        height: 42px;
        font-size: ${theme.font.sizes.large};
        padding: ${theme.spacings.medium};
        padding-left: 0;
      `}
    }
  `}
`
