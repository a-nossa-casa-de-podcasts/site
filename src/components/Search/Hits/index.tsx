import Link from 'next/link'
import * as S from './styles'

export type HitProps = {
  id: number
  slug: string
  title: string
  subTitle: string
  number: number
  releaseDate: string
  description: string
  podcast: string
  author: string
}

export type HitsListProps = {
  hits: HitProps[]
}

const HitList = ({ hits }: HitsListProps) =>
  hits ? (
    <S.Wrapper>
      <h3>Resultados</h3>
      {hits.map((item, index) => (
        <S.List key={index}>
          <div>
            <Link href={`/${item.slug}`} passHref>
              <a target="_self" title={item.title}>
                <em>{item.podcast && <span>{item.podcast}</span>}</em>
                {item.author && (
                  <span className="light-emphasis">
                    &nbsp;por <em>{item.author}</em>
                  </span>
                )}
                <h3>
                  {item.title} {item.number && <span>#{item.number}</span>}
                </h3>
                {item.subTitle && <p>{item.subTitle}</p>}
                {item.description && <p>{item.description}</p>}
              </a>
            </Link>
          </div>
        </S.List>
      ))}
    </S.Wrapper>
  ) : (
    <S.Wrapper>
      <h3>Nenhum resultado encontrado</h3>
    </S.Wrapper>
  )

export default HitList
