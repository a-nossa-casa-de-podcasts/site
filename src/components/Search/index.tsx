import * as S from './styles'

const Search = () => (
  <S.Wrapper>
    <S.SearchInputWrapper>
      <S.SearchInput
        aria-label="search"
        placeholder="Busque por podcast ou tema"
        type="text"
      ></S.SearchInput>
    </S.SearchInputWrapper>
    <S.SearchResultsWrapper>
      <div>
        <span>
          <b>Radiofonias</b> Internéticas
        </span>
        <h3>
          <b>Radiofonias</b> Internéticas - S1 E1
        </h3>
        <p>
          Nosso primeiro episódio é conduzido pela @heloisapr que nos leva a um
          papo com Aline Valek, escritora, contadora de histórias e podcaster do
          Bobagens Imperdíveis. Prepara o fone, clica no link da nossa bio e se
          jogue nessa escuta! O Radiofonias é um lugar de experimento, escuta &
          de ideias.
        </p>
      </div>
      <div>
        <span>
          <b>Radiofonias</b> Internéticas
        </span>
        <h3>
          <b>Radiofonias</b> Internéticas - S1 E1
        </h3>
        <p>
          Nosso primeiro episódio é conduzido pela @heloisapr que nos leva a um
          papo com Aline Valek, escritora, contadora de histórias e podcaster do
          Bobagens Imperdíveis. Prepara o fone, clica no link da nossa bio e se
          jogue nessa escuta! O Radiofonias é um lugar de experimento, escuta &
          de ideias.
        </p>
      </div>
    </S.SearchResultsWrapper>
  </S.Wrapper>
)

export default Search
