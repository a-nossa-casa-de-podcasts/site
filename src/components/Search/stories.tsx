import { Story, Meta } from '@storybook/react/types-6-0'
import Search from '.'

export default {
  title: 'Design System/Organisms/Search',
  component: Search
} as Meta

export const Default: Story = () => <Search />

Default.parameters = {
  layout: 'fullscreen'
}
