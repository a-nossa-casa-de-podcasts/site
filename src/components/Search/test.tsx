import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Search from '.'

describe('<Search />', () => {
  it('should render the search', () => {
    const { container } = renderWithTheme(<Search />)

    expect(screen.getByLabelText('search')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
