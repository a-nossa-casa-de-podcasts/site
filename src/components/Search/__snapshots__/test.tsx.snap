// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render the search 1`] = `
.c0 {
  background-color: #1C77C6;
  height: 100vh;
  overflow: hidden;
  padding: 2.4rem;
  padding-bottom: 0;
  display: grid;
  grid-template-rows: 60px auto minmax(10px,60px);
}

.c1 {
  position: relative;
  width: 100%;
}

.c2 {
  font-size: 1.5rem;
  font-weight: regular;
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  width: 100%;
  background: transparent;
  color: #FFFFFF;
  border: none;
  border-bottom: 2px solid white;
  padding: 0.8rem;
  padding-left: 0;
  outline: none;
}

.c3 {
  font-size: 1.8rem;
  color: #FFFFFF;
  height: 80vh;
  overflow-x: hidden;
  overflow-y: auto;
}

.c3 div {
  padding-right: 1.6rem;
  padding-bottom: 1.6rem;
  margin-bottom: 2.4rem;
  border-bottom: 1px solid #439FEF;
}

.c3 b {
  color: #0AF9B5;
}

.c3 span {
  font-weight: 700;
  font-size: 1.3rem;
}

.c3 h3 {
  font-weight: 700;
  font-size: 1.5rem;
}

.c3 p {
  font-weight: regular;
  font-size: 1.3rem;
}

@media (min-width:768px) {
  .c0 {
    grid-template-rows: 100px auto minmax(10px,60px);
    padding: 14rem;
  }
}

@media (min-width:768px) {
  .c2 {
    height: 42px;
    font-size: 4.0rem;
    padding: 3.2rem;
    padding-left: 0;
  }
}

@media (min-width:768px) {
  .c3 div {
    padding-right: 0;
  }

  .c3 span {
    font-weight: 700;
    font-size: 1.5rem;
  }

  .c3 h3 {
    font-weight: 700;
    font-size: 1.8rem;
  }

  .c3 p {
    font-weight: regular;
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <input
      aria-label="search"
      class="c2"
      placeholder="Busque por podcast ou tema"
      type="text"
    />
  </div>
  <div
    class="c3"
  >
    <div>
      <span>
        <b>
          Radiofonias
        </b>
         Internéticas
      </span>
      <h3>
        <b>
          Radiofonias
        </b>
         Internéticas - S1 E1
      </h3>
      <p>
        Nosso primeiro episódio é conduzido pela @heloisapr que nos leva a um papo com Aline Valek, escritora, contadora de histórias e podcaster do Bobagens Imperdíveis. Prepara o fone, clica no link da nossa bio e se jogue nessa escuta! O Radiofonias é um lugar de experimento, escuta & de ideias.
      </p>
    </div>
    <div>
      <span>
        <b>
          Radiofonias
        </b>
         Internéticas
      </span>
      <h3>
        <b>
          Radiofonias
        </b>
         Internéticas - S1 E1
      </h3>
      <p>
        Nosso primeiro episódio é conduzido pela @heloisapr que nos leva a um papo com Aline Valek, escritora, contadora de histórias e podcaster do Bobagens Imperdíveis. Prepara o fone, clica no link da nossa bio e se jogue nessa escuta! O Radiofonias é um lugar de experimento, escuta & de ideias.
      </p>
    </div>
  </div>
</main>
`
